pypeit/core/findobj_skymask.py

```diff
@@ -147,7 +147,1159 @@ def create_skymask(sobjs, thismask, slit_left, slit_righ, box_rad_pix=None, trim
+                                                             sigma_lower=5.0,sigma_upper=5.0)
+            # ToDO assign this to sobjs_align for use in the extraction
+            SNR_arr[iord,iobj] = med_sn
+            sobjs_align[indx][0].ech_snr = med_sn
+            # For hand extractions
+            slitfracpos_arr[iord,iobj] = sobjs_align[indx][0].SPAT_FRACPOS
+
+    # Purge objects with low SNR that don't show up in enough orders, sort the list of objects with respect to obj_id
+    # and orderindx
+    keep_obj = np.zeros(nobj,dtype=bool)
+    # Empty specobjs object to hold the final objects
+    sobjs_trim = specobjs.SpecObjs()
+    # objids are 1 based so that we can easily asign the negative to negative objects
+    iobj_keep = 1
+    iobj_keep_not_hand = 1
+
+    ## Loop over objects from highest SNR to lowest SNR. Apply the S/N constraints. Once we hit the maximum number
+    # objects requested exit, except keep the hand apertures that were requested.
+    isort_SNR_max = np.argsort(np.median(SNR_arr,axis=0))[::-1]
+    for iobj in isort_SNR_max:
+        hand_ap_flag = hand_flag[iobj]
+        SNR_constraint = (SNR_arr[:,iobj].max() > max_snr) or (
+            np.sum(SNR_arr[:,iobj] > min_snr) >= nabove_min_snr)
+        nperorder_constraint = (iobj_keep-1) < nperorder
+        if (SNR_constraint and nperorder_constraint) or hand_ap_flag:
+            keep_obj[iobj] = True
+            ikeep = sobjs_align.ECH_OBJID == uni_obj_id[iobj]
+            sobjs_keep = sobjs_align[ikeep].copy()
+            sobjs_keep.ECH_OBJID = iobj_keep
+            sobjs_keep.OBJID = iobj_keep
+            sobjs_trim.add_sobj(sobjs_keep[np.argsort(sobjs_keep.ECH_ORDERINDX)])
+            iobj_keep += 1
+            if not hand_ap_flag:
+                iobj_keep_not_hand += 1
+        else:
+            if not nperorder_constraint:
+                msgs.info('Purging object #{:d}'.format(iobj) +
+                          ' since there are already {:d} objects automatically identified '
+                          'and you set nperorder={:d}'.format(iobj_keep_not_hand-1, nperorder))
+            else:
+                msgs.info('Purging object #{:d}'.format(iobj) + ' which does not satisfy max_snr > {:5.2f} OR min_snr > {:5.2f}'.format(max_snr, min_snr) +
+                ' on at least nabove_min_snr >= {:d}'.format(nabove_min_snr) + ' orders')
+
+
+    nobj_trim = np.sum(keep_obj)
+
+    if nobj_trim == 0:
+        msgs.warn('No objects found')
+        sobjs_final = specobjs.SpecObjs()
+        return sobjs_final
+
+    # TODO JFH: We need to think about how to implement returning a maximum number of objects, where the objects
+    # returned are the highest S/N ones. It is a bit complicated with regards to the individual object finding and then
+    # the linking that is performed above, and also making sure the hand apertures don't get removed.
+    SNR_arr_trim = SNR_arr[:,keep_obj]
+
+    sobjs_final = sobjs_trim.copy()
+    return sobjs_final
```

Comment:
Skip the middle man: `return sobjs_trim.copy()`?

Reply:
agreed

Reply:
and no `.copy()` necessary